File: src/api/developer.ts
import { post } from '@/utils/request'

// 申请成为开发者
export function register<T>(data: Developer.RegisterRequest) {
  return post<T>({
    url: '/developer/register',
    data,
  })
}

// 获取当前用户的开发者信息
export function getInfo<T>() {
  return post<T>({
    url: '/developer/getInfo',
  })
}

// 更新开发者信息
export function updateMyInfo<T>(data: Developer.UpdateMyInfoRequest) {
  return post<T>({
    url: '/developer/update',
    data,
  })
}

// 检查是否是开发者
export function checkIsDeveloper<T>() {
  return post<T>({
    url: '/developer/checkIsDeveloper',
  })
}

File: service/router/panel/developer.go
package panel

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitDeveloperRouter(router *gin.RouterGroup) {
	developerApi := api_v1.ApiGroupApp.ApiPanel.DeveloperApi

	r := router.Group("", middleware.LoginInterceptor)
	{
		r.POST("developer/register", developerApi.Register)
		r.POST("developer/getInfo", developerApi.GetInfo)
		r.POST("developer/update", developerApi.Update)
		r.POST("developer/checkIsDeveloper", developerApi.CheckIsDeveloper)
	}
}
